src/hooks/use-theme.js

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useCallback, useState } from 'react';
+
+import useLocalStorage from './use-local-storage';
+
+const useTheme = () => {
+  const [storedTheme, setStoredTheme] = useLocalStorage('selected_theme', 'system');
+  const [selectedTheme, setSelectedTheme] = useState(storedTheme);
+
+  const isSystemThemeDark = () => window.matchMedia('(prefers-color-scheme: dark)').matches === true;
+
+  useEffect(() => {
+    if (
+      (selectedTheme === 'system' && isSystemThemeDark()) || (selectedTheme === 'dark')
+    ) {
+      document.body.classList.add('dark');
+    } else {
+      document.body.classList.remove('dark');
+    }
+  }, [selectedTheme]);
+
+  const toggleTheme = useCallback(() => {
+    let newTheme = 'system';
+    if (selectedTheme === 'system') {
+      newTheme = isSystemThemeDark() ? 'light' : 'dark';
+    } else {
+      newTheme = selectedTheme === 'dark' ? 'light' : 'dark';
+    }
+
+    setStoredTheme(newTheme);
+    setSelectedTheme(newTheme);
+  }, [selectedTheme, setStoredTheme, setSelectedTheme]);
+
+  return { selectedTheme, toggleTheme };
+};
+
+export default useTheme;
```

Comment:
I think we can simplify this a bit - there no need for state AND localStorage - localStorage should suffice

something like this could work
```
import { useEffect } from 'react';

import useLocalStorage from './use-local-storage';

const DARK = 'dark';
const LIGHT = 'light';

const useTheme = () => {
  const isSystemThemeDark = () => window.matchMedia('(prefers-color-scheme: dark)').matches === true;

  const [storedTheme, setStoredTheme] = useLocalStorage('selected_theme', isSystemThemeDark ? DARK : LIGHT);

  useEffect(() => {
    if (
      (storedTheme === DARK)
    ) {
      document.body.classList.add(DARK);
    } else {
      document.body.classList.remove(DARK);
    }
  }, [storedTheme]);

  const toggleTheme = () => {
    setStoredTheme(storedTheme === DARK ? LIGHT : DARK);
  };

  return { storedTheme, toggleTheme };
};

export default useTheme;
```